CMakeLists.txt

```diff
@@ -97,6 +97,15 @@ option(TON_USE_TSAN "Use \"ON\" to enable ThreadSanitizer." OFF)
 option(TON_USE_UBSAN "Use \"ON\" to enable UndefinedBehaviorSanitizer." OFF)
 set(TON_ARCH "native" CACHE STRING "Architecture, will be passed to -march=")
 
+#BEGIN M1 support
+EXECUTE_PROCESS( COMMAND uname -m COMMAND tr -d '\n' OUTPUT_VARIABLE ARCHITECTURE )
+
+if (ARCHITECTURE MATCHES "arm64")
+    set(TON_ARCH "apple-m1")
```

Comment:
Hi, it's not guaranteed if the architecture is `arm64` then cpu is `apple-m1`. I think we should also check that `CMAKE_SYSTEM_NAME` equals `Darwin`.